<p>
    Unused parameters are misleading. Whatever the value passed to such parameters is, the behavior will be the same.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
void DoSomething(int a, int b) // "b" is unused
{
  Compute(a);
}

void DoSomething2(int a) // value of "a" is unused
{
  a = 10;
  Compute(a);
}
</pre>

<h2>Compliant Solution</h2>
<pre>
void DoSomething(int a)
{
  Compute(a);
}

void DoSomething2()
{
  var a = 10;
  Compute(a);
}
</pre>

<h2>Exceptions</h2>
<p>
    <code>virtual</code>, <code>override</code> methods and interface implementations are ignored.
</p>
<pre>
override void DoSomething(int a, int b) // no issue reported on b
{
  Compute(a);
}
</pre>

<p>
    Furthermore, the <code>this</code> parameter of extension methods is also ignored.
</p>
<pre>
public static class Extensions
{
  public static void MyHelper(this HtmlHelper helper) //no issue reported here
  {
    // no use of helper here
  }
}
</pre>

<p>
    Methods that have attributes defined on them are also ignored.
</p>
<pre>
public class MyDto
{
  public string Name { get; set; }

  [OnDeserialized]
  private void OnDeserialized(StreamingContext context)
  {
    // ...
  }
}
</pre>

<h2>See</h2>

<ul>
    <li>MISRA C++:2008, 0-1-11</li>
    <li>MISRA C:2012, 2.7</li>
</ul>
