<p>
    While it is technically correct to assign to parameters from within method bodies, it is better to use temporary variables to store intermediate results.
    This rule will typically detect cases where a constructor parameter is assigned to itself instead of a field of the same name, i.e. when <code>Me</code> was forgotten.
    Allowing parameters to be assigned to also reduces the code readability as developers will not be able to know whether the original parameter or some temporary variable is being accessed without going through the whole method.
    Moreover, some developers might also expect assignments of method parameters to be visible from callers, which is not the case and can confuse them.
    All parameters should be treated as <code>ReadOnly</code>.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
Module Module1
    Sub Foo(ByVal a As Integer)
        a = 42                  ' Noncompliant
    End Sub
End Module
</pre>

<h2>Compliant Solution</h2>
<pre>
Module Module1
    Sub Foo(ByVal a As Integer)
        Dim tmp = a
        tmp = 42                ' Compliant
    End Sub
End Module
' or
Module Module1
    Sub Foo(ByRef a As Integer)
        a = 42                  ' Compliant - it is a ByRef parameter
    End Sub
End Module
</pre>

<h2>See</h2>
<ul>
    <li>MISRA C:2012, 17.8 - A function parameter should not be modified</li>
</ul>
