<p>
    Having two <code>Cases</code> in the same <code>Select</code> statement or branches in the same <code>If</code> structure
    with the same implementation is at best duplicate code, and at worst a coding error. If the same logic is truly needed for
    both instances, then in an <code>If</code> structure they should be combined, or for a <code>Select</code>, one should fall
    through to the other.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
Select i
  Case 1
    doSomething()
  Case 2
    doSomethingDifferent()
  Case 3  ' Noncompliant; duplicates case 1's implementation
    doSomething()
  Case Else:
    doTheRest()
End Select

If a >= 0 AndAlso a < 10 Then
  doTheThing()
ElseIf a >= 10 AndAlso a < 20 Then
  doTheOtherThing()
ElseIf a >= 20 AndAlso a < 50
  doTheThing() ' Noncompliant; duplicates first condition
Else
  doTheRest();
End If
</pre>

<h2>Compliant Solution</h2>
<pre>
Select i
  Case 1, 3
    doSomething()
  Case 2
    doSomethingDifferent()
  Case Else
    doTheRest()
End Select

If (a >= 0 AndAlso a < 10) OrElse (a >= 20 AndAlso a < 50) Then
  doTheThing()
ElseIf a >= 10 AndAlso a < 20 Then
  doTheOtherThing()
Else
  doTheRest();
End If
</pre>
or
<pre>
Select i
  Case 1
    doSomething()
  Case 2
    doSomethingDifferent()
  Case 3
    doTheThirdThing()
  Case Else
    doTheRest()
End Select

If a >= 0 AndAlso a < 10 Then
  doTheThing()
ElseIf a >= 10 AndAlso a < 20 Then
  doTheOtherThing()
ElseIf a >= 20 AndAlso a < 50
  doTheThirdThing()
Else
  doTheRest();
End If
</pre>
