<p>
    Property getters should be simple operations that are always safe to call. If exceptions need to be thrown, it is best to convert the property to a method.
</p>
<p>
    It is valid to throw exceptions from indexed property getters and from property setters, which are not detected by this rule.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
class MyClass
{
  public int Foo
  {
    get
    {
      throw new NotImplementedException();  //Non-Compliant
    }
    set
    {
      // ... some code ...
    }
  }
}
</pre>

<h2>Compliant Solution</h2>

<pre>
class MyClass
{
  public void SetFoo(int value)
  {
    // ... some code ...
  }
}
</pre>
