<p>
    Namespaces with no lines of code clutter a project and should be removed.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
namespace MyEmptyNamespace // Noncompliant
{

}
</pre>

<h2>See</h2>
<ul>
    <li><a href="https://www.securecoding.cert.org/confluence/x/OYIyAQ">CERT, MSC07-CPP.</a> - Detect and remove dead code</li>
</ul>
